AdyenCard/Utilities/CardType/CardBrandProvider.swift

```diff
@@ -105,22 +105,18 @@ internal final class CardBrandProvider: AnyCardBrandProvider {
         
         let localApiClient = self.apiClient ?? APIClient(environment: self.environment)
         
-        do {
-            try cardPublicKeyProvider.fetch { [weak self] result in
-                guard let self = self else { return }
-                
-                switch result {
-                case let .success(publicKey):
-                    let binLookupService = BinLookupService(publicKey: publicKey,
-                                                            apiClient: localApiClient)
-                    self.privateBinLookupService = binLookupService
-                    success(binLookupService)
-                case let .failure(error):
-                    failure?(error)
-                }
+        cardPublicKeyProvider.fetch { [weak self] result in
+            guard let self = self else { return }
+
+            switch result {
+            case let .success(publicKey):
+                let binLookupService = BinLookupService(publicKey: publicKey,
+                                                        apiClient: localApiClient)
+                self.privateBinLookupService = binLookupService
+                success(binLookupService)
+            case let .failure(error):
+                failure?(error)
```

Comment:
maybe failure can be a non-optional parameter. or, even better, `fetchBinLookupService` can accept only one closure, of type `Result<BinLookupService, Swift.Error>`